extensions/compositions/compositions.tsx

```diff
@@ -7,50 +6,36 @@ import { PropTable } from '@teambit/documenter.ui.property-table';
 import { Tab, TabContainer, TabList, TabPanel } from '@teambit/panels';
 import { Collapser } from '@teambit/staged-components.side-bar';
 import { EmptyBox } from '@teambit/staged-components.empty-box';
-import { gql } from 'apollo-boost';
 import head from 'lodash.head';
-import R from 'ramda';
-import React, { useContext, useEffect, useReducer, useState } from 'react';
+import React, { useContext, useEffect, useState, useRef } from 'react';
 
 import { Composition } from './composition';
 import styles from './compositions.module.scss';
 import { ComponentComposition } from './ui';
 import { CompositionsPanel } from './ui/compositions-panel/compositions-panel';
-
-const GET_COMPONENT = gql`
-  query($id: String!) {
-    getHost {
-      getDocs(id: $id) {
-        properties {
-          name
-          description
-          required
-          type
-          default: defaultValue {
-            value
-          }
-        }
-      }
-    }
-  }
-`;
+import { useProperties } from './use-properties';
 
 export function Compositions() {
   const component = useContext(ComponentContext);
-  // const compositions = useCompositions();
   const [selected, selectComposition] = useState(head(component.compositions));
-  const { data } = useQuery(GET_COMPONENT, {
-    variables: { id: component.id._legacy.name },
-  });
-  const properties = R.path(['getHost', 'getDocs', 'properties'], data);
+  const selectedRef = useRef(selected);
+  selectedRef.current = selected;
+
+  const properties = useProperties(component.id);
+
   // reset selected composition when component changes.
   // this does trigger renderer, but perf seems to be ok
   useEffect(() => {
```

Comment:
improvement to composition sidebar:
* keep same composition if it has the same identifier,
* toggle sidebar when adding composition or removing the last one